src/runtime_src/core/pcie/linux/pcidrv_xclmgmt.h

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+#ifndef _XCL_PCIDRV_XCLMGMT_H_
+#define _XCL_PCIDRV_XCLMGMT_H_
+
+#include "pcidrv.h"
+
+namespace xrt_core { namespace pci {
+
+class drv_xclmgmt : public drv
+{
+public:
+  const std::string
```

Comment:
```suggestion
  std::string
```
const here doesn't do anything at all.

Reply:
It means that the content of the string can't be changed, no?

Reply:
Return-by-value, which is correct here, but the rvalue will be assigned to some lvalue which caller can change.

Reply:
This string contains the name of the driver/module and should never be changed, even by the caller. So, I can return const string for this?

Reply:
You are returning a copy, the caller owns the copy and can do with it what they want.
std::string mycpy = name();  // will be fine, and mycpy can be changed
auto mycpy = name();  // will be fine, but auto does not capture cv-qualifiers so same as above